packages/ibm-products/src/components/BasicHeader/BasicHeader.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright IBM Corp. 2020, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { Breadcrumb, BreadcrumbItem } from '@carbon/react';
+import PropTypes from 'prop-types';
+
+import React, { useCallback, useEffect } from 'react';
+import cx from 'classnames';
+import { pkg } from '../../settings';
+import pconsole from '../../global/js/utils/pconsole';
+
+const blockClass = `${pkg.prefix}--basic-header`;
+const componentName = 'BasicHeader';
+/**
+ * The BasicHeader is not public and only used internally by CreateFullPage.
+ *
+ * Component varieties:
+ * - Header with Breadcrumbs
+ * - Header with Title
+ * - Header with Breadcrumbs and Title
+ *
+ *
+ * The component will throw a warning message if neither a title nor breadcrumbs are provided
+ * since it requires at least one of them.
+ * */
+const BasicHeader = ({
+  breadcrumbs,
+  className,
+  title,
+  noTrailingSlash,
+  ...rest
+}) => {
+  const warnIfNoTitleOrBreadcrumbs = useCallback(() => {
+    if (!title && !breadcrumbs?.length) {
+      pconsole.error(
+        `Warning: You have tried using a ${componentName} component without specifying a title nor breadcrumbs props`
+      );
+    }
+  }, [breadcrumbs, title]);
+
+  useEffect(() => {
+    warnIfNoTitleOrBreadcrumbs();
+  }, [title, breadcrumbs, warnIfNoTitleOrBreadcrumbs]);
+
+  return (
+    <header className={cx(blockClass, className)} {...rest}>
+      {breadcrumbs?.length > 0 && (
+        <Breadcrumb
```

Comment:
May be worth using [`BreadcrumbWithOverflow`](https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-internal-breadcrumbwithoverflow--default). Looks like Carbon also allows overflow as a child of `BreadcrumbItem` too now ([Breadcrumb with overflow menu](https://react.carbondesignsystem.com/?path=/story/components-breadcrumb--breadcrumb-with-overflow-menu)), but think `BreadcrumbWithOverflow` might be the simpler update?

Reply:
I am now using the BreadcrumbWithOverflow, but my gut feeling is that the component is a bit over-complicated in terms of how its written and what it renders in the DOM (since it renders duplicate breadcrumbs in the DOM where one is set to display none and the other is visible based on the container size)..and that made it a bit challenging to fix my tests..
I would suggest adding a TODO or something similar to consider refactoring the BreadcrumbWithOverflow component.